### PR TITLE
10790 686 chap 2

### PR DESCRIPTION
**Pull request by lbai001**
#### veteran condition: 
screencapture-localhost-3001-disability-benefits-apply-dependents-form-686c-dependents-veteran-information-2018-06-08-03_12_12
#### non veteran condition:
screencapture-localhost-3001-disability-benefits-apply-dependents-form-686c-dependents-veteran-information-2018-06-08-03_12_58

### CODE REVIEW

File: src/applications/disability-benefits/686/helpers.jsx [hunk 1]
```diff
@@ -104,14 +98,13 @@ export const otherRelationshipDescription = (
...
-export function isNotVeteran(item) {
-  return get('view:relationshipToVet', item) !== '1';
+export function isVeteran(item) {
+  const relationship = get('view:relationshipToVet', item);
+  return relationship && relationship === '1';
 }
 
-export function isSpouse(item) {
-  return get('application.claimant.relationshipToVet', item) === '2';
+export function isNotVeteran(item) {
```
Comment (jbalboni) [inline src/applications/disability-benefits/686/helpers.jsx]: You could have this function call `isVeteran` and negate the output, to reduce the duplication.

File: src/applications/disability-benefits/686/config/form.js [hunk 2]
```diff
@@ -190,6 +193,114 @@ const formConfig = {
...
+                'ui:required': formData => isNotVeteran(formData)
+              },
+              middle: { 'ui:title': 'Veteran’s middle name' },
+              last: {
+                'ui:title': 'Veteran’s last name',
+                'ui:required': formData => isNotVeteran(formData)
+              },
+              suffix: { 'ui:title': 'Veteran’s suffix' }
+            }),
+            veteranSocialSecurityNumber: {
```
Comment (jbalboni) [inline src/applications/disability-benefits/686/config/form.js]: Typically we do something like this to add things to common components:

```js
veteranSocialSecurityNumber: _.merge(ssnUI, {
  'ui:options': {
    updateSchema: (form, schema) => {
      if (isVeteran(form)) {
        return _.merge(schema, {
          title: 'Your Social Security number'
        });
      }
      return _.merge(schema, {
        title: 'Veteran’s Social Security number'
      });
    }
  },
  'ui:required': formData => !formData['view:noSSN'],
});
```

It makes it less error prone if we adjust the SSN definition later.

File: src/applications/disability-benefits/686/config/form.js [hunk 3]
```diff
@@ -190,6 +193,114 @@ const formConfig = {
...
+              'ui:options': {
+                updateSchema: (form, schema) => {
+                  if (isVeteran(form)) {
+                    return _.merge(schema, {
```
Comment (jbalboni) [inline src/applications/disability-benefits/686/config/form.js]: I don't think you need the `_.merge` here (and in the other similar locations), by default the forms library code will do this for you and merge the object you return with the current schema.

**Review by jbalboni — COMMENTED** [review]
Had a couple comments, but looks good so far.

**Comment by emilywaggoner** [comment]
Hey @lbai001! Looking awesome. I have a couple of questions/notes:

- For both conditions, is it possible to only show the "VA File number" field if they check the preceeding checkbox ("I don't have a Social Security number")?
- In the Non-Veteran condition would you mind changing the checkbox to say "I don't know the Veteran's Social Security number"? (Instead of "I don't _have_")

Thanks Annie!

**Comment by lbai001** [comment]
@emilywaggoner 
screen shot 2018-06-11 at 2 52 05 am
screen shot 2018-06-11 at 2 52 13 am

**Review by jbalboni — APPROVED** [review]
Couple minor comments, but LGTM.

**Review by emilywaggoner — APPROVED** [review]
All looks good to me, Annie. Thank you!

**Pull request merged**